### DISCUSSION TITLE
caldav backend doesn't work

### DISCUSSION THREAD

**Discussion by orlovnv** [Z-Hub/Z-Push#123] [Q&A] [answered]
/usr/local/lib/z-push/include/z_caldav.php:675 Undefined array key "DAV::href" (2)
/usr/local/lib/z-push/include/z_caldav.php:675 Trying to access array offset on value of type null (2)
/usr/local/lib/z-push/include/z_caldav.php:676 Undefined array key ""
/usr/local/lib/z-push/include/z_caldav.php:676 Trying to access array offset on value of type null (2)

**Comment by matidau** [comment]
It would help to know what caldav you are using? This message may be related to no calendars for the user, or a calendar isn't returning a href element.

**Reply by orlovnv** [reply]
I try with SOGo, define('CALDAV_PATH', '/SOGo/dav/%u/Calendar'); define('CALDAV_PERSONAL', 'personal'); and in iPhone I see Unknown cal, but cal name is test, when I add another one with name work it shows me 2 unknown calendar

when I try define('CALDAV_PATH', '/SOGo/dav/%u/Calendar/personal'); it change from unknown to Calendar, but doesn't show any info in it

**Reply by matidau** [reply]
Can you try creating a calendar called `personal` or set `CALDAV_PERSONAL` to work or test?

**Reply by orlovnv** [reply]
I add personal the same problem 2 calendar with name Calendar

**Reply by orlovnv** [reply]
/usr/local/lib/z-push/include/z_caldav.php:801 Undefined property: CalDAVClient::$xmlnodes (2)

**Reply by orlovnv** [reply, marked as answer]
I did it, thank you very mach
